.github/scripts/attack.py

```
@@ -336,8 +336,72 @@ def attack_test():
+
+def wait_for_broker_quiesce(timeout_sec: float = 60.0) -> bool:
+    """
+    Block until a fresh client completes a QoS-1 subscribe/publish/receive
+    round trip promptly, proving the broker has drained the stress backlog.
+    """
+    log = mk_logger("SETTLE")
+    deadline = time.time() + timeout_sec
+    while time.time() < deadline:
+        connected = threading.Event()
+        got = threading.Event()
+        c = new_client(f"SETTLE_{random.randint(1000, 9999)}")
```

Ruff (0.15.21)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)